### PR TITLE
Improve issue reference on commit

### LINKED ISSUE #6664: Feature Request: Cross-link commits mentioning other repos
- Gitea version (or commit ref):8d01b98
- Git version:
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
Refering to this pull request #6273, on this pull request say able to link issue to another repo
I tried in here https://try.gitea.io/stani0502/coba.
i tried commit and put url of the issue from different repo to link this issue https://try.gitea.io/stani0502/test/issues/1.
But is not working.
Let me know how the correct way to do that.

...

## Screenshots
image
image

### PR DESCRIPTION
**Pull request by mrsdizzie**
Allow commits to properly reference issues in other repositories and also to close/reopen those issues if user has code permission. Should match Github behavior described here: https://help.github.com/en/articles/closing-issues-using-keywords

Fixes #6664

This also fixes a bug I noticed where if you reference another repo in the form of ``` user/repo#1``` Gitea will currently just match the ```#1``` so if the user ```gitea```commits a file that says ```Closes gogs/gogs#1```to the repo ```gitea/gitea```it will actually close issue ```#1``` on the gitea repo

### CODE REVIEW

File: models/action.go [hunk 1]
```diff
@@ -192,6 +193,24 @@ func (a *Action) GetRepoLink() string {
 	return "/" + a.GetRepoPath()
 }
 
+// GetRepositoryFromMatch returns a *Repository from a username/repo string
+func GetRepositoryFromMatch(match string) (*Repository, error) {
+	parts := strings.Split(match, "/")
```
Comment (zeripath) [inline models/action.go]: You know you could have already had the regexp split this for you?

**Review by zeripath — APPROVED** [review]
Seems reasonable, just a comment that you could have already had the regexp do the splitting for you.

Finally, just make certain that usernames and repository names don't have different first letter requirements from the rest of their names - I don't think they do but it might be reasonable to check.

File: models/action.go (hunk 1 above)
Comment (mrsdizzie) [reply models/action.go]: ah yea, that would make more sense. Do you think its worth it to ditch the separate function for this and just call GetRepositoryByOwnerAndName  etc... directly then? This function isn't doing much I just wasn't sure if calling GetRepositoryByOwnerAndName and checking for errors three separate times in a row was too much code duplication.

Comment (zeripath) [reply models/action.go]: If I was to refactor I would look at whether you could reduce the amount of code duplication or split up updateissuecommit. That's a very long function and removing your getrepositorybymatch function would only make it longer.

Comment (mrsdizzie) [reply models/action.go]: Agree, will keep it in for now to avoid this being even longer. Adjusted to have matching done in regex itself thx.

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**